### DISCUSSION TITLE
brew install gerbera on macOS Mojave fails

### DISCUSSION THREAD

**Discussion by man00ka** [gerbera/gerbera#2369] [Q&A]
Hello,

after brew runs `cmake .. -DWITH_CURL=YES -DWITH_JS=YES -DWITH_TAGLIB=YES -DWITH_AVCODEC=YES -DWITH_EXIF=YES -DWITH_SYSTEMD=NO -DWITH_INOTIFY=NO -DWITH_MYSQL=NO -DWITH_FFMPEGTHUMBNAI
`

I get this output:
```Last 15 lines from /Users/Manu/Library/Logs/Homebrew/gerbera/01.cmake:
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Looking for __atomic_fetch_add_4 in atomic
-- Looking for __atomic_fetch_add_4 in atomic - not found
-- Looking for C++ include filesystem
-- Looking for C++ include filesystem - not found
CMake Error at cmake/modules/FindFilesystem.cmake:227 (message):
  Cannot Compile simple program using std::filesystem
Call Stack (most recent call first):
  CMakeLists.txt:312 (find_package)
```

unfortunately I cannot retrieve the install log from my /tmp.

Can anyone tell me what this is about?

Thanks in advance!

**Comment by KarlStraussberger** [comment]
Looks like the wrong C++ version. Gerbera needs a compiler which supports at least cpp-2017

**Reply by man00ka** [reply]
I uninstalled Command Line Tools for X-Code and clean-installed CLT for X-Code 11.3.1

now I get this message after `brew install gerbera`:
```Last 15 lines from /Users/Manu/Library/Logs/Homebrew/gerbera/01.cmake:
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED - Failed
-- Performing Test CXX_FILESYSTEM_STDCPPFS_NEEDED
-- Performing Test CXX_FILESYSTEM_STDCPPFS_NEEDED - Failed
-- Performing Test CXX_FILESYSTEM_CPPFS_NEEDED
-- Performing Test CXX_FILESYSTEM_CPPFS_NEEDED - Failed
CMake Error at cmake/modules/FindFilesystem.cmake:227 (message):
  Cannot Compile simple program using std::filesystem
Call Stack (most recent call first):
  CMakeLists.txt:312 (find_package)
```

I tried `brew install gerbera --cc=gcc` after installing gcc-11.2. However, same error message.

**Reply by man00ka** [reply]
unfortunately again I am not able to retrieve `/tmp/gerbera-20220104-99589-qqgy4n/gerbera-1.7.0/build/CMakeFiles/CMakeOutput.log` as mentioned in the `stdout` after failed install. There is no such folder `gerbera-...` in `/tmp`

**Reply by KarlStraussberger** [reply]
Which version of gerbera are you building? From the output I see 1.7.0 - current is 1.9.2 (or you even go for the git-master)

**Reply by man00ka** [reply]
Thanks for your reply!

Well, from the gerbera homepage I was sent to here `https://github.com/gerbera/homebrew-gerbera/` where I followed the instructions to add the tap to homebrew and ran `brew install gerbera` assuming brew to install the latest stable version. Which version would you suggest me to install instead?

Thanks again for you help!

**Reply by KarlStraussberger** [reply]
So we need a volunteer to update the code to 1.9.2 and maybe supply some recipe for gerbera-master. :smile:

**Comment by whyman** [comment]
The 1.7.0 tap builds fine here, but this is on Big Sur

```==> Installing gerbera/gerbera/gerbera
==> cmake .. -DWITH_CURL=YES -DWITH_JS=YES -DWITH_TAGLIB=YES -DWITH_AVCODEC=YES -DWITH_EXIF=YES -DWITH_SYSTEMD=NO -DWITH_INOTIFY=NO -DWITH_MYSQL=NO -DWITH_FFMPEGTHUMBNAILER=YES -DCMAKE_CXX_FLAGS="-stdlib=libc++" -DCMAKE_CXX_
==> make install
🍺  /usr/local/Cellar/gerbera/1.7.0: 88 files, 7.2MB, built in 1 minute 26 seconds```

**Reply by man00ka** [reply]
which version of CLT for X-Code do you have installed? Maybe I can install the same version and try again.

**Reply by whyman** [reply]
```
Clang: 12.0.5 build 1205
macOS: 11.6.1-x86_64
CLT: 12.5.0.22.9
```

**Reply by man00ka** [reply]
Since `CLT 11.3.1` (which I have installed) ist the latest version for `macOS 10.14` and I am hesitant to upgrade to 10.15 (which might be able to handle the gerbera compilation(?)) is there anything else I could try to make the compilation work on `macOS 10.14.6` (using a MacBook Pro early 2013, if that helps).

- Is there maybe another compiler I could try? I've already tried gcc-11.2 which brought up the same error messages about failed tests.
- Is there a way to get the error logs?  According to output of my last try, they should have been placed here:
```
-- Configuring incomplete, errors occurred!
See also "/tmp/gerbera-20220104-99589-qqgy4n/gerbera-1.7.0/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/gerbera-20220104-99589-qqgy4n/gerbera-1.7.0/build/CMakeFiles/CMakeError.log".
```
I checked immediately  after the try. However, nothings there. `gerbera-20220104-99589-qqgy4n` doesn't exist. Actually nothing related to gerbera is written to `/tmp`

Maybe log files could shed some light on why the tests failed in the first place.

**Reply by whyman** [reply]
You could try using the llvm/clang from homebrew
